lib/fastlane/plugin/aws_device_farm/actions/aws_device_farm_action.rb

```diff
@@ -430,6 +430,20 @@ def self.wait_for_run(project, run)
         )
         puts ""
 
+        Dir.mkdir('fastlane/test_output') unless Dir.exist?('fastlane/test_output')
+        Dir.mkdir("fastlane/test_output/#{params[:name]}") unless Dir.exist?("fastlane/test_output/#{params[:name]}")
+        out_file = File.new("fastlane/test_output/#{params[:name]}/report.junit", "w")
```

Comment:
consider

building a string, like

```ruby
  output = ""
  output << "<?xml version='1.0' encoding='UTF-8'?>"
  output << "<testsuite name='#{params[:name]}'>"

  job.jobs.each do | j |
    output << "<testcase classname='#{params[:device_pool]}' name='#{j.name}' time='#{j.device_minutes.total*60}'>"
   if j.result != "PASSED"
         out << "<failure type='Failed'>Test failed on #{j.name}</failure>")
     end
  out << "</testcase>"

 end
out << "</testsuite>"
File.write('/path/to/file', 'Some glorious content')

```


this way, you only write to disk once.
if it is getting a problem with memory, because `output` gets to large you can also consider doing it.

```ruby
File.write('/path/to/file', 'Some glorious content') # first write
File.write('/path/to/file', 'Some glorious content', "a") # append to file.
```

this way you atleast get rid of the `File.new` thing



`File.write('/path/to/file', 'Some glorious content')` - and you can get rid of the `File.new`